src/Integration.UnitTests/LocalServices/ErrorListInfoBarController.QualityProfileBackgroundProcessorTests.cs

```diff
@@ -0,0 +1,386 @@
+
+        #region Tests
+        [TestMethod]
+        public void QualityProfileBackgroundProcessor_ArgChecks()
+        {
+            // Act + Verify
+            Exceptions.Expect<ArgumentNullException>(() => 
+                new ErrorListInfoBarController.QualityProfileBackgroundProcessor(null));
+        }
+
+        [TestMethod]
+        public void QualityProfileBackgroundProcessor_LifeCycle()
+        {
+            // Setup
+            var testSubject = this.GetTestSubject();
+
+            // Verify
+            Assert.IsNotNull(testSubject.TokenSource);
+            Assert.AreNotEqual(CancellationToken.None, testSubject.TokenSource.Token);
+
+            // Act
+            testSubject.Dispose();
+
+            // Verify
+            Exceptions.Expect<ObjectDisposedException>(() => testSubject.TokenSource.Cancel());
+        }
+
+        [TestMethod]
+        public void QualityProfileBackgroundProcessor_QueueCheckIfUpdateIsRequired_ArgChecks()
+        {
+            // Setup
+            var testSubject = this.GetTestSubject();
+
+            // Act + Verify
+            Exceptions.Expect<ArgumentNullException>(() => testSubject.QueueCheckIfUpdateIsRequired(null));
+        }
+
+        [TestMethod]
+        public void QualityProfileBackgroundProcessor_QueueCheckIfUpdateIsRequired_NoFilteredProjects()
+        {
+            // Setup
+            var testSubject = this.GetTestSubject();
+            this.projectSystem.Projects = new Project[] { new ProjectMock("project.proj") };
+            this.projectSystem.FilteredProjects = null;
+
+            // Act
+            testSubject.QueueCheckIfUpdateIsRequired(this.AssertIfCalled);
+
+            // Verify
+            this.outputWindow.AssertOutputStrings(0);
+        }
+
+        [TestMethod]
+        public void QualityProfileBackgroundProcessor_QueueCheckIfUpdateIsRequired_NoSolutionBinding()
+        {
+            // Setup
+            var testSubject = this.GetTestSubject();
+            var project1 = new ProjectMock("validProject1.csproj");
+            project1.SetCSProjectKind();
+            var project2 = new ProjectMock("validProject2.csproj");
+            project2.SetCSProjectKind();
+            this.projectSystem.FilteredProjects = new[] { project1, project2 };
+
+            // Act
+            testSubject.QueueCheckIfUpdateIsRequired(this.AssertIfCalled);
+
+            // Verify
+            this.outputWindow.AssertOutputStrings(0);
+        }
+
+        [TestMethod]
+        public void QualityProfileBackgroundProcessor_QueueCheckIfUpdateIsRequired_NoProfiles_RequiresUpdate()
+        {
+            // Setup
+            var testSubject = this.GetTestSubject();
+            var project1 = new ProjectMock("validProject1.csproj");
+            project1.SetCSProjectKind();
+            var project2 = new ProjectMock("validProject2.csproj");
+            project2.SetCSProjectKind();
+            this.projectSystem.FilteredProjects = new[] { project1, project2 };
```

Comment:
There's a lot of duplication in the setup section across the tests which makes it harder to see what the important differences are. For example, the code to set the filtered projects is repeated numerous times, but it looks like the only important detail is the language of the project. A helper method like _SetFilteredProjects(params languageType)_ would remove the duplication and emphasise the important details.